test/distributed/test_distributed.py

```diff
@@ -2750,33 +2755,84 @@ def test_ddp_grad_div_uneven_inputs(self):
         grad_scale = 50
         rank = self.rank
         model = nn.Linear(dim, dim, bias=False)
-        inp = torch.ones(batch, dim).to(self.rank) * grad_scale
+        inp = torch.ones(batch, dim, device=self.rank) * grad_scale
         net = torch.nn.parallel.DistributedDataParallel(
             model.cuda(rank), device_ids=[self.rank], bucket_cap_mb=1
         )
         n_iters = 1 if self.rank == 0 else 2
-        from torch.nn.parallel.distributed import join
 
-        with join(net) as ddp_join:
+        with net.join(net):
```

Comment:
Why do we need to pas `net` to join? Shouldn't `net.join()` be sufficient?

Reply:
My bad, it should indeed just be `net.join()` as we use in the other tests. I think the reason that this passes is because `net` is taken to be the boolean `enable` argument, and `if enable` just checks for truthiness. So maybe we should assert that if anything is passed into `join`, it must be a boolean.